lib/debts/service.rb

```diff
@@ -2,10 +2,57 @@
 
 module Debts
   class Service < Common::Client::Base
+    include Common::Client::Concerns::Monitoring
+
     configuration Debts::Configuration
 
+    STATSD_KEY_PREFIX = 'api.debts'
+
     def get_letters(body)
-      GetLettersResponse.new(perform(:post, 'letterdetails/get', body).body)
+      with_monitoring_and_error_handling do
+        GetLettersResponse.new(perform(:post, 'letterdetails/get', body).body)
+      end
+    end
+
+    private
+
+    def with_monitoring_and_error_handling
+      with_monitoring(2) do
+        yield
+      end
+    rescue => e
+      handle_error(e)
+    end
+
+    def save_error_details(error)
+      Raven.tags_context(
+        external_service: self.class.to_s.underscore
+      )
```

Comment:
Do we want to include tests to ensure we're sending the expected details to Raven?

Reply:
yeah that's a good idea. I'll add that and then resolve this convo.

Reply:
Added